fdbcli/TenantCommands.actor.cpp

```diff
@@ -651,6 +651,92 @@ ACTOR Future<bool> tenantRenameCommand(Reference<IDatabase> db, std::vector<Stri
 	return true;
 }
 
+ACTOR Future<bool> tenantLockCommand(Reference<IDatabase> db, std::vector<StringRef> tokens) {
+	state UID uid;
+	state Reference<ITransaction> tr;
+	state StringRef name;
+	state Key nameKey;
+	state TenantLockState desiredLockState;
+	state int uidIdx;
+	if (tokens[1] == "lock"_sr && (tokens.size() < 3 || tokens.size() > 5)) {
+		fmt::print("Usage: tenant lock <NAME> [w|rw] [UID]\n\n");
+		fmt::print("Locks a tenant with for read-write or read-only with a given UID.\n");
+		fmt::print("generate one. By default a read-write lock is created.\n");
+		fmt::print(" If no UID is passed, fdbcli will UID has to be a 16-byte number represented in hex.");
+		return false;
+	} else if (tokens[1] == "unlock"_sr && tokens.size() != 4) {
+		fmt::print("Usage: tenant unlock <NAME> <UID>\n\n");
+		return false;
+	}
+	name = tokens[2];
+	nameKey = tenantMapSpecialKeyRange.begin.withSuffix(name);
+	if (tokens[1] == "unlock"_sr) {
+		uidIdx = 3;
+		desiredLockState = TenantLockState::UNLOCKED;
+	} else {
+		uidIdx = 4;
+		if (tokens.size() > 3) {
+			if (tokens[3] == "w"_sr) {
+				desiredLockState = TenantLockState::READ_ONLY;
+			} else if (tokens[3] == "rw"_sr) {
+				desiredLockState = TenantLockState::LOCKED;
+			} else {
+				fmt::print(stderr, "Invalid lock type \"{}\"\n", tokens[3]);
+				return false;
+			}
+		} else {
+			desiredLockState = TenantLockState::LOCKED;
+		}
+	}
+	if (tokens.size() > uidIdx) {
+		try {
+			uid = UID::fromStringThrowsOnFailure(tokens[uidIdx].toString());
```

Comment:
To make this more easy to use, we should also report the lock state and lock ID in `TenantMapEntry::toJson()` and in `tenantGetCommand`